unitTests/testStackArray.cpp

```diff
@@ -16,61 +16,265 @@
  *~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  */
 
+// Source includes
 #include "StackBuffer.hpp"
-#include "gtest/gtest.h"
 #include "Array.hpp"
+#include "testUtils.hpp"
 
+// TPL includes
+#include <gtest/gtest.h>
 
-using namespace LvArray;
+// System includes
+#include <cmath>
+
+namespace LvArray
+{
+namespace testing
+{
 
 template< typename T, int NDIM, int MAX_SIZE >
 using stackArray = StackArray< T, NDIM, camp::make_idx_seq_t< NDIM >, int, MAX_SIZE >;
 
+LVARRAY_HOST_DEVICE
+int toLinearIndex( int const i )
+{ return i; }
+
+LVARRAY_HOST_DEVICE
+int toLinearIndex( int const i, int const j )
+{ return i + 10 * j; }
+
+LVARRAY_HOST_DEVICE
+int toLinearIndex( int const i, int const j, int const k )
+{ return i + 10 * j + 100 * k; }
 
-TEST( StackArray, allocate1d )
+constexpr int pow( int const base, int const exponent )
+{ return exponent == 0 ? 1 : base * pow( base, exponent - 1 ); }
+
+template< typename NDIM_T >
+class StackArrayTest : public ::testing::Test
 {
-  stackArray< int, 1, 100 > array1d;
-  array1d.resize( 10 );
+public:
+  static constexpr int NDIM = NDIM_T::value;
+
+  void resize()
+  {
+
+    init();
+
+    m_array.resize( 4 );
+
+    forValuesInSliceWithIndices( m_array.toSlice(), []( int const value, auto const ... indices )
+    {
+      EXPECT_EQ( value, toLinearIndex( indices ... ) );
+    } );
+
+    int dims[ NDIM ];
+    for( int i = 0; i < NDIM; ++i )
+    { dims[ i ] = i + 1; }
+
+    m_array.resize( NDIM, dims );
+
+    for( int i = 0; i < NDIM; ++i )
+    { EXPECT_EQ( m_array.size( i ), i + 1 ); }
+
+    forValuesInSliceWithIndices( m_array.toSlice(), []( int & value, auto const ... indices )
+    {
+      value = toLinearIndex( indices ... );
+    } );
+
+    ASSERT_DEATH_IF_SUPPORTED( m_array.resize( 2 * CAPACITY ), "" );
+  }
 
-  for( int i=0; i<10; ++i )
+protected:
+
+  void init()
   {
-    array1d[i] = i;
-    ASSERT_TRUE( array1d.data()[i] == i );
+    int dims[ NDIM ];
+
+    for( int i = 0; i < NDIM; ++i )
+    { dims[ i ] = 8; }
+
+    m_array.resize( NDIM, dims );
```

Comment:
In the flow solvers, there are a lot of places where we use the `stackArray` as: 
```c++
stackArray1d< real64, MAX_NUM_FACES > property( numFaces ); 
```
and with this PR tested on Ascent this is not compiling in a function marked `GEOSX_HOST_DEVICE` (and sorry if I misunderstood something). But, everything works well with a `resize`:
```c++ 
stackArray1d< real64, MAX_NUM_FACES > property;
property.resize( numFaces );
``` 
I am fine with both, but perhaps we should completely disallow the first option (on host and on device), or maybe allow it on device as well.

Reply:
Thanks, I just forgot to to decorate that method.

Reply:
@francoishamon it should work now.